### PR TITLE
#164693948 Validate length of article input fields and refactor code

### PR DESCRIPTION
**Pull request by shaolinmkz**
#### What does this PR do?
Have the character length of the article input fields validated.
Have some sections of the code-base refactored.

#### Description of Task to be completed?
Having the following task below completed:
1. Minimum character length of the title should be 5.
2. Maximum character length of the title should be 200.
3. Minimum character length of description should be 5.
4. Maximum character length of description should be 1000.
5. Minimum character length of the body should be 10.
6. Maximum character length of the body should be infinite.
7. Disable last-name and first-name field from being required because of twitter.
8. Change GET all bookmark route from `/api/articles/bookmarks` to `/api/bookmarks`
9. Remove unused lines of code.
10. Write more test.
11. Validate input fields for reset password route.

#### How should this be manually tested?
1. Clone this branch.
2. Run `npm install` to install the dependencies.
3. Run `npm test` to run the tests.

#### What are the relevant pivotal tracker stories?
[#164693948](https://pivotaltracker.com/story/show/164693948)

### CODE REVIEW

**Review by Chrismarcel — COMMENTED** [review]
Amazing job. However, I noticed some slight errors in some of your test descriptions. Kindly, ensure to fix. Thank you

**Comment by uniqueayo1988** [comment]
I observed some endpoints were renamed, good. Other from the test descriptions pointed out by Chris, your code looks good to me.

File: server/database/models/user.js [hunk 1]
```diff
@@ -3,11 +3,11 @@ export default (sequelize, DataTypes) => {
     {
       firstname: {
         type: DataTypes.STRING,
-        allowNull: false
+        allowNull: true
```
Comment (andela-gike) [inline server/database/models/user.js:6]: @shaolinmkz, is there a reason you set the last name and first name to allow Null.

**Review by andela-gike — CHANGES_REQUESTED** [review]

File: server/database/models/user.js (hunk 1 above)
Comment (shaolinmkz) [reply server/database/models/user.js:6]: It was due to the Twitter social login, Twitter doesn't return `firstname` and `lastname` just `display name` and `username`. So if I make it required it returns an error when data from twitter is being saved to the database.
But it is required for normal signup... There is a middleware that validates that.

**Review by andela-gike — APPROVED** [review]

**Pull request merged**